app/views/shop/products/_form.html.haml

```diff
@@ -1,54 +1,58 @@
-
-        %product{"ng-show" => "Products.loading"}
-          .row.summary
-            .small-12.columns.text-center
-              = t :products_loading
+      .small-12.columns
+        .footer-pad.small-12.columns
           .row
-            .small-12.columns.text-center
-              %img.spinner{ src: "/assets/spinning-circles.svg" }
-
-        %div{"ng-show" => "Products.products.length == 0 && !Products.loading"}
-          .row.summary
-            .small-12.columns
-              %p.no-results
-                = t :search_no_results_html, query: "<strong>{{query}}</strong>".html_safe
-      .row
-        .small-12.columns
-          %form{action: main_app.cart_path}
-            %i.ofn-i_011-spinner.cart-spinner{"ng-show" => "Cart.dirty"}
-            %input.small.button.primary.right.add_to_cart{type: :submit, value: "{{ Cart.dirty ? '#{t(:products_updating_cart)}' : (Cart.empty() ? '#{t(:products_cart_empty)}' : '#{t(:products_edit_cart)}' ) }}", "ng-disabled" => "Cart.dirty || Cart.empty()", "ng-class" => "{ dirty: Cart.dirty }" }
+            .medium-12.large-10.columns
+              = render partial: "shop/products/search_feedback"
+
+              %div.pad-top{ "infinite-scroll" => "loadMore()", "infinite-scroll-distance" => "1", "infinite-scroll-disabled" => 'Products.loading' }
+                %product.animate-repeat{"ng-controller" => "ProductNodeCtrl", "ng-repeat" => "product in Products.products track by product.id", "id" => "product-{{ product.id }}"}
+                  = render "shop/products/summary"
+                  %shop-variant{variant: 'variant', "ng-repeat" => "variant in product.variants | orderBy: ['name_to_display','unit_value'] track by variant.id", "id" => "variant-{{ variant.id }}", "ng-class" => "{'out-of-stock': !variant.on_demand && variant.on_hand == 0}"}
+
+                %product{"ng-show" => "Products.loading"}
+                  .row.summary
+                    .small-12.columns.text-center
+                      = t :products_loading
+                  .row
+                    .small-12.columns.text-center
+                      %img.spinner{ src: "/assets/spinning-circles.svg" }
+
+                %div{"ng-show" => "Products.products.length == 0 && !Products.loading"}
+                  .row.summary
+                    .small-12.columns
+                      %p.no-results
+                        = t :search_no_results_html, query: "<strong>{{query}}</strong>".html_safe
+              .row
+                .small-12.columns
+                  %i.ofn-i_011-spinner.cart-spinner{"ng-show" => "Cart.dirty"}
+                  %input.small.button.primary.right.add_to_cart{type: :submit, value: "{{ Cart.dirty ? '#{t(:products_updating_cart)}' : (Cart.empty() ? '#{t(:products_cart_empty)}' : '#{t(:products_edit_cart)}' ) }}", "ng-disabled" => "Cart.dirty || Cart.empty()", "ng-class" => "{ dirty: Cart.dirty }" }
+
+            .hide-for-medium-down.large-2.columns
```

Comment:
I like when the partials are small.
I'd extract what's above this line up to line 13 to a partial: _products_list for example
and what's below this line until the end to another partial: _product_list_filters

Reply:
Is it okay to leave some of these refactoring bits until later? I've made 2 other PRs that are on top of this one (not in code review yet), and they both make various changes that I think would make this quite messy to pick apart and re-apply if I do start extracting these bits to partials in this PR. Actually I think I've already extracted some bits like this in the next PRs...

Reply:
yeah, it was just a suggestion.